packages/runtime/src/runtime.ts

```diff
@@ -459,23 +487,47 @@ export class Runtime {
   }
 
   /**
-   * creates new application and returns application id
-   * @param flags SSCDeployment flags
-   * @param payFlags Transaction parameters
+   * create new application and returns application id
+   * @deprecated `deployApp` should be used instead.
    * @param approvalProgram application approval program
    * @param clearProgram application clear program
+   * @param flags SSCDeployment flags
+   * @param payFlags Transaction parameters
    * @param debugStack: if passed then TEAL Stack is logged to console after
    * each opcode execution (upto depth = debugStack)
    * NOTE - approval and clear program must be the TEAL code as string (not compiled code)
    */
   addApp (
+    approvalProgram: string, clearProgram: string,
     flags: AppDeploymentFlags, payFlags: types.TxParams,
+    debugStack?: number
+  ): DeployedAppTxReceipt {
+    this.addCtxAppCreateTxn(flags, payFlags);
+    this.ctx.debugStack = debugStack;
+    const txReceipt = this.ctx.deployApp(flags.sender.addr, flags, approvalProgram, clearProgram, 0);
+
+    this.store = this.ctx.state;
+    return txReceipt;
```

Comment:
why not just use `this.deployApp`?

Reply:
Oh, That is great. Thanks for your suggestion (y).